src/codesyncd/validators/tab_validator.ts

```diff
@@ -0,0 +1,69 @@
+/*
+class TabValidator:
+	validateRepoID()
+	validate_yml_file()
+	group_data()
+*/
+
+import { REQUIRED_KEYS_TAB_FILE_YML, TAB_SIZE_LIMIT, VSCODE } from "../../constants";
+import { ITabYML } from "../../interface";
+import { CodeSyncLogger } from "../../logger";
+import { generateSettings } from "../../settings";
+import { readYML } from "../../utils/common";
+import { ConfigUtils } from "../../utils/config_utils";
+import { removeFile } from "../../utils/file_utils";
+import { UserState } from "../../utils/user_utils";
+
+export class TabValidator {
+
+	settings: any;
+	configJSON: any;
+	constructor() {
+		this.settings = generateSettings();
+        this.configJSON = readYML(this.settings.CONFIG_PATH);
+
+	}
+
+	validateRepo(tabFile: ITabYML, tabData: string, filePath: string, repoId: number) {
+		const config_utils = new ConfigUtils();
+		const repo_path = config_utils.getRepoPathByRepoId(repoId);
+		const configRepo = this.configJSON.repos[repo_path];
+		const repoEmail = configRepo.email;
```

Comment:
We are checking `if (!configRepo)if (!configRepo)` after this line, it should be after that check I think